enumerables.rb

```diff
@@ -0,0 +1,160 @@
+def hash_engine(&block)
+  0.upto(length - 1) { |index| block.call(keys[index], values[index]) }
+  self
+end
+
+def array_engine(&block)
+  0.upto(length - 1) { |index| block.call(self[index], index) }
+  self
+end
+
+def range_engine(&block)
+  array = [*self]
+  0.upto(array.length - 1) { |index| block.call(array[index], index) }
+  array
+end
+
+def engine_select_block_check(&block)
+  return to_enum unless block_given?
+
+  if self.class == Hash
+    hash_engine(&block)
+  elsif self.class == Array
+    array_engine(&block)
+  elsif self.class == Range
+    range_engine(&block)
+  end
+end
+
+def convert_to_array
+  array = if self.class == Range
+            [*self]
+          else
+            self
+          end
+  array
+end
+
+def class_check(arg)
+  if arg.class == NilClass
+    my_select { |x| x }
+  elsif arg.class == Regexp
+    my_select { |x| x.match?(arg) }
+  elsif arg.class == String
+    my_select { |x| x.include?(arg) }
+  elsif arg.class == Integer
+    my_select { |x| x == arg }
+  elsif arg.class == Class
+    numeric_inclusion(arg)
+  end
+end
+
+def numeric_inclusion(arg_class)
+  if arg_class == Numeric
+    my_select { |arg| true if arg.class == Integer || arg.class == Float || arg.class == Complex }
+  else
+    my_select { |x| x.class == arg_class }
+  end
+end
+
+def t_f_test(arg, &block)
+  var = block.call(arg)
+  return true unless var.nil? || var == false
+end
+
+module Enumerable
+  def my_each(&block)
+    return to_enum unless block_given?
+
+    engine_select_block_check(&block)
+    self
+  end
+
+  def my_each_with_index(_arg = nil, &block)
+    return to_enum unless block_given?
+
+    my_each(&block)
+    self
+  end
+
+  def my_select(&block)
+    return to_enum unless block_given?
+
+    hash_query = {}
+    array_query = []
+    if self.class == Hash
+      0.upto(length - 1) do |ind|
+        hash_query[keys[ind]] = values[ind] unless block.call(keys[ind], values[ind]) == false
+      end
+    else
+      engine_select_block_check { |val| array_query << val if t_f_test(val, &block) }
+    end
+    array_query.empty? ? hash_query : array_query
+  end
```

Comment:
# my_select 
1. [ ] Returns an array containing all elements of enum for which the given block returns a true value::range
### Test 👉 `expect(range.my_select(&block)).to eq(range.select(&block))`
|Expected|Got|
|:---:|:---:|
|` []`|`{}`|